engines/dfc_provider/app/controllers/dfc_provider/api/catalog_items_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Controller used to provide the API products for the DFC application
+module DfcProvider
+  module Api
+    class CatalogItemsController < DfcProvider::Api::BaseController
+      def index
+        render json: current_user, serializer: DfcProvider::PersonSerializer
+      end
+
+      def show
+        render json: variant, serializer: DfcProvider::CatalogItemSerializer
+      end
+
+      private
+
+      def variant
+        @variant ||=
+          Spree::Variant.
+            joins(product: :supplier).
+            where('enterprises.id' => current_enterprise.id).
+            find(params[:id])
```

Comment:
I think this controller is supposed to show products which are sold/distributed and not just produced/supplied, is that right?

Reply:
About data mapping, nothing is fixed, I mostly mapped as I could in order to return data for tests. We also need to use the same data source than on the OFN API, we will need to call same services or objects. As @luisramos0 mentioned it, we can't afford to maintain 2 different APIs, DFC API should call OFN logic the same way than the OFN API.